homeassistant/helpers/entity.py

```diff
@@ -425,6 +428,20 @@ def _name_translation_key(self) -> str | None:
             f".{self.translation_key}.name"
         )
 
+    def _name_placeholders(self, name: str) -> str:
+        """Process possible placeholders in entity name."""
+        tuples = list(string.Formatter().parse(name))
+        if tuples[0][1] is None:
+            return name
+        if (translation_placeholders := self.translation_placeholders) is not None:
+            if TYPE_CHECKING:
+                assert isinstance(translation_placeholders, Mapping)
+            try:
+                return name.format(**translation_placeholders)
+            except KeyError as err:
+                raise HomeAssistantError("Missing placeholder %s" % err) from err
+        raise HomeAssistantError("Missing property _attr_translation_placeholders")
```

Comment:
The exception text is misleading, we end up here if the property `translation_placeholders` is None.

Reply:
No longer relevant